### PR TITLE
redis aof backup and fetch, basic integration tests

### PR DESCRIPTION
**Pull request by kronwerk**
### Database name
redis

# Pull request description
adding possibility to make backups with AOF files (7.0+ only)

### Describe what this PR fix
none

### Please provide steps to reproduce (if it's a bug)
none

### Please add config and wal-g stdout/stderr logs for debug purpose
none

### CODE REVIEW

File: docker/redis/redis.conf [hunk 1]
```diff
@@ -265,7 +265,7 @@ dir /var/lib/redis/
 ################################# REPLICATION #################################
 
 # Master-Slave replication. Use slaveof to make a Redis instance a copy of
-# another Redis server. A few things to understand ASAP about Redis replication.
+# another Redis server. A few things to understand ASAP about Redis replsupervisorication.
```
Comment (VGoshev) [inline docker/redis/redis.conf:268]: Looks like typo for me

File: internal/databases/redis/archive/backup.go [hunk 2]
```diff
@@ -121,13 +150,17 @@ type BackupMeta struct {
 	User           interface{}
 	StartTime      time.Time
 	FinishTime     time.Time
+	BackupType	   string
```
Comment (VGoshev) [inline internal/databases/redis/archive/backup.go:153]: NIT: Looks like too much spaces here

File: internal/databases/redis/archive/backup.go [hunk 3]
```diff
@@ -121,13 +150,17 @@ type BackupMeta struct {
...
+	Version 	string
 }
 
 type RedisMetaConstructor struct {
 	ctx       context.Context
 	folder    storage.Folder
 	meta      BackupMeta
 	permanent bool
+	backupType string
```
Comment (VGoshev) [inline internal/databases/redis/archive/backup.go:162]: The same here, looks like need to reformat (un `go fmt` or something)

File: internal/databases/redis/archive/backup_test.go [hunk 4]
```diff
@@ -17,6 +17,8 @@ func TestBackup_PrintableFields(t *testing.T) {
 		BackupSize:      200000,
 		Permanent:       true,
 		UserData:        []string{"a", "b", "c"},
+		Version: "4.5.4",
```
Comment (VGoshev) [inline internal/databases/redis/archive/backup_test.go:20]: `go fmt`

File: internal/databases/redis/archive/backup_test.go [hunk 5]
```diff
@@ -64,6 +66,18 @@ func TestBackup_PrintableFields(t *testing.T) {
 			Value:       "true",
 			PrettyValue: nil,
 		},
+		{
+			Name: "backup_type",
+		    PrettyName: "Backup type",
```
Comment (VGoshev) [inline internal/databases/redis/archive/backup_test.go:71]: `go fmt` as well %)

**Review by VGoshev — CHANGES_REQUESTED** [review]
In general looks good for me, but could you fix one typo and run `go fmt` (or something you like) on files you've changed to reformat them, please.

Also definitely some behave tests needed for aof backups but they can be added later

**Comment by kronwerk** [comment]
> In general looks good for me, but could you fix one typo and run `go fmt` (or something you like) on files you've changed to reformat them, please.
> 
> Also definitely some behave tests needed for aof backups but they can be added later

some files were not touched by "make fmt" due to insufficient folder depth in wildcard - added one more level to internal and added "go fmt ./..." just in case also

**Review by VGoshev — APPROVED** [review]

**Review by munakoiso — APPROVED** [review]

File: internal/databases/redis/archive/fs.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+type Folder struct {
+	Path     string
+	fileMode fs.FileMode
+}
+
+func CreateFolder(path string, fileMode fs.FileMode) *Folder {
+	return &Folder{
+		Path:     path,
+		fileMode: fileMode,
+	}
```
Comment (ostinru) [inline internal/databases/redis/archive/fs.go:17]: Name is misleading - it doesn't create folder =(

**Review by ostinru — APPROVED** [review]
Good enough to be merged.

File: internal/databases/redis/archive/fs.go (hunk 6 above)
Comment (kronwerk) [reply internal/databases/redis/archive/fs.go:17]: thx, gonna change this in next patch

**Review by perekalov — APPROVED** [review]

**Review by teem0n — APPROVED** [review]

**Review by Apelsin234 — APPROVED** [review]

**Review by KhurtinDN — APPROVED** [review]

**Review by debebantur — APPROVED** [review]

**Pull request merged**